cucumber_cpp/library/cucumber_expression/Ast.cpp

```diff
@@ -0,0 +1,143 @@
+#include "cucumber_cpp/library/cucumber_expression/Ast.hpp"
+#include <cctype>
+#include <map>
+#include <ranges>
+#include <stdexcept>
+#include <string>
+#include <utility>
+#include <variant>
+#include <vector>
+
+namespace cucumber_cpp::library::cucumber_expression
+{
+    namespace
+    {
+        const std::map<char, TokenType>& DemarcationMap()
+        {
+            static const std::map<char, TokenType> demarcationMap = {
+                { '/', TokenType::alternation },
+                { '{', TokenType::beginParameter },
+                { '}', TokenType::endParameter },
+                { '(', TokenType::beginOptional },
+                { ')', TokenType::endOptional },
+            };
+
+            return demarcationMap;
+        }
+
+        const std::map<TokenType, std::string>& TokenTypeMap()
+        {
+            static const std::map<TokenType, std::string> tokenTypeMap = {
+                { TokenType::alternation, "alternation" },
+                { TokenType::beginParameter, "beginParameter" },
+                { TokenType::endParameter, "endParameter" },
+                { TokenType::beginOptional, "beginOptional" },
+                { TokenType::endOptional, "endOptional" },
+            };
+            return tokenTypeMap;
+        }
+    }
+
+    std::string Node::Text()
+    {
+        return static_cast<const Node*>(this)->Text();
+    }
+
+    std::string Node::Text() const
+    {
+        if (std::holds_alternative<std::string>(children))
+            return std::get<std::string>(children);
+
+        if (!std::get<std::vector<Node>>(children).empty())
+        {
+            std::string text{ std::get<std::vector<Node>>(children).front().Text() };
+            for (const auto& child : std::get<std::vector<Node>>(children) | std::views::drop(1))
+                text += child.Text();
+            return text;
+        }
+
+        return {};
+    }
+
+    std::vector<Node>& Node::Children()
+    {
+        return std::get<std::vector<Node>>(children);
+    }
+
+    const std::vector<Node>& Node::Children() const
+    {
+        return std::get<std::vector<Node>>(children);
+    }
+
+    bool Token::IsEscapeCharacter(char ch)
+    {
+        return ch == '\\';
+    }
+
+    TokenType Token::TypeOf(char ch)
+    {
+        if (std::isspace(ch) != 0)
+            return TokenType::whiteSpace;
+
+        if (DemarcationMap().contains(ch))
+            return DemarcationMap().at(ch);
+
+        return TokenType::text;
+    }
+
+    bool Token::CanEscape(char ch)
+    {
+        if (std::isspace(ch) != 0)
```

Comment:
@daantimmer Maybe replace this line too with `if (std::isspace(static_cast<unsigned char>(ch)) != 0)` ?

Reply:
or `std::isspace(ch, std::locale())`